### ISSUE TITLE
taxonomy: allow creation of taxons in ODM2 linked from external systems

### ISSUE THREAD

**Issue by RmStorm** [NIVANorge/odm2-postgres-api#84]
The goal is to give researchers the ability to 'import' species from external systems. Current candidate systems are:
[Artsdatabanken](https://www.artsdatabanken.no/)
[WORMS](http://www.marinespecies.org/)
[Algaebase](https://www.algaebase.org/)

The idea is _not_ that researchers have access to all species in these systems when they are inserting observations. The idea _is_ that researchers add specific species to the taxonomy table in ODM2 and that in this process they can choose to link to an external system.

So, for example, a researcher encounters a species they have not seen before. Through their expertise they track down the exact species that it is. They find it in, say, Algaebase.  Now to register the observation they first need to make this species available in out system. In their specific user interface, [e.g. begroing](https://begroing.test.niva.no/#register_species), they add the species. In there they will have a button that says, 'import species from external system'. Or something similar.

After they have found their species the app needs to hit an endpoint in ODM2 that inserts the species in the `TaxonomicClassifiers` table and that also inserts a link in `TaxonomicClassifierExternalIdentifiers`. This means that the external systems we want to support have to be available in `ExternalIdentifierSystems`.

Any PR resolving this issue will need to create new endpoints for this functionality and add metadata for the external systems we decide to support.

The biggest open questions are regarding which data to synchronize over from the possible external systems.

We need at least:
- Common name
- Identifier in external systems

We need almost certainly:
- Author of the species

We quite likely also need to sync:
- synonyms
- direct parent in the hierarchy

**Comment by DagHjermann** [comment]
I think one should sync all of those. The synonyms would be needed as that would enable us to collect all samples of the same species in our own data. The parent would be needed to collect all samples from a given genus / family / order. 

A couple of issues: 
(1) Which name should be used when (say) WORMS and Artsdatabanken conflicts? (Wenche mentioned that WORMS tends to be more up-to-date than ADB.) Let the researcher decide, or have a preference for one of them? 
(2) A related issue: taxonomy tends to change from time to time. A quite common change, for instance, is that a species changes genus, i.e. the "first name" changes. For instance, snowy owl was until recently called _Nyctea scandiaca_ (the only species of this genus) but then genetic evidence showed that it is closely related to the _Bubo_ owls, so now many (but in this case, not all) specialists call it _Bubo scandiaca_.  Should we then change the original name used for the sample, or should we keep the old name even though we start using the new one? I think I vote for the latter.

**Comment by jeved** [comment]
@RmStorm , I think we should get all fields, but probably the last one can wait. It depends a bit how you store the species name. If it is genus + species name, you should be OK, if you only store the second part of the species name, then you would need a link to the genus. I guess the former is the best solution.

To the second comment by @DagHjermann: I think we all agree on that the observation is linked to the actual name chosen by the researcher, and that we keep the "old" name

**Comment by DagHjermann** [comment]
If we don't store/sync the parent taxa, we might have some problem with "sp." taxa with the same name when we search for them in Artsdatabanken? For instance if some sample is stored with "Trichoptera sp." then a search in Artsdatabanken will return both the order Trichoptera (caddisflies/vårfluer) and a genus Trichoptera (a genus in "vinterfluer"). There are probably more such homonyms in ADB and probably even more in WORMS. It is probably always quite easy to say which of them the researcher meant, but not so easy to do _automatically_ using ADB's API (which I guess is the plan?).

**Comment by jeved** [comment]
Yes @DagHjermann , I think you are quite right, but I just meant this could wait, especially if this may impose a lot of work. Most species will possibly have one ore more unique identifiers:
- Combination of name and author
- Adb latin name id, or maybe some other taxonomic database id, like worms-id?
- Combination of name and rank (yes, we forgot rank, I guess that would be necessary

But maybe we it is something I haven't thouht of. 

Anyway, @RmStorm , i think **rank** would be a good idea at some stage as well.
